### PR TITLE
Add test cases to ensure proper synchronization

### PR DESCRIPTION
**Pull request by nakabonne**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/piped/executor/kubernetes/sync_test.go [hunk 1]
```diff
@@ -19,17 +19,166 @@ import (
...
+				}(),
+				config: &config.KubernetesDeploymentSpec{
+					QuickSync: config.K8sSyncStageOptions{
+						AddVariantLabelToSelector: true,
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
```
Comment (nghialv) [inline pkg/app/piped/executor/kubernetes/sync_test.go:173]: nit: `tt` -> `tc`, `tests` -> `testcases`

Comment (nakabonne) [reply pkg/app/piped/executor/kubernetes/sync_test.go:173]: Right, hopefully the code genereted by Jetbrains Goland can configurable...

**Comment by nakabonne** [comment]
Applied!

**Pull request merged**